File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    var answer = [];
    let index = 10000;
    let num = 0;
    
    for(let target of targets){
        for(let i=0; i<target.length; i++){
            for(let key of keymap){
                if(key.includes(target[i]) && index > key.indexOf(target[i]) ){
                    index = key.indexOf(target[i]);
                }
            }
            if(index !== 10000){
                num += index+1;
                index = 10000;
            }else{
                num =0;
                break;
            }
        }
        if(num == 0){
            answer.push(-1);
        }
        else{
        answer.push(num);
        num = 0;
        }
    }
    return answer;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];

        for (int i = 0; i < targets.length; i++) {
            int total = 0;

            for (int j = 0; j < targets[i].length(); j++) {
                char target = targets[i].charAt(j);
                int min = Integer.MAX_VALUE;

                for (int k = 0; k < keymap.length; k++) {
                    for (int l = 0; l < keymap[k].length(); l++) {

                        if (keymap[k].charAt(l) == target) {
                            min = Math.min(min, l + 1);
                            break;
                        }

                    }
                }
                if (min == Integer.MAX_VALUE) {
                    total = -1;
                    break;
                }
                total += min;
            }
            answer[i] = total;
        }
        return answer;
    }
}